datafusion/src/optimizer/constant_folding.rs

```diff
@@ -210,6 +210,74 @@ impl<'a> ExprRewriter for Simplifier<'a> {
                         right,
                     },
                 },
+                Operator::Or => match (left.as_ref(), right.as_ref()) {
+                    (
+                        Expr::Literal(ScalarValue::Boolean(l)),
+                        Expr::Literal(ScalarValue::Boolean(r)),
```

Comment:
The  `ScalarValue` == `ScalarValue` case is covered by `ConstEvaluator` -- so while it is not incorrect to add it here, I also don't think it is necessary (and might be confusing as it would imply `ConstEvaluator` can't handle this.